driver/src/main/scala/api/api.scala

```diff
@@ -509,12 +525,13 @@ class MongoDriver(systemOption: Option[ActorSystem] = None) {
    */
   def connection(nodes: Seq[String], options: MongoConnectionOptions = MongoConnectionOptions(), authentications: Seq[Authenticate] = Seq.empty, nbChannelsPerNode: Int = 10, name: Option[String] = None): MongoConnection = {
     val props = Props(new MongoDBSystem(nodes, authentications, options)())
-    val mongosystem = if (name.isDefined) system.actorOf(props, name = name.get) else system.actorOf(props)
-    val connection = new MongoConnection(system, mongosystem, options)
-    this.synchronized {
-      _connections = connection :: _connections
+    val mongosystem = name match {
+      case Some(nm) => system.actorOf(props, nm);
+      case None => system.actorOf(props, "Connection-" +  + MongoDriver.nextCounter)
     }
-    connection
+    implicit val timeout: Timeout = Timeout(10, TimeUnit.SECONDS)
+    val connection = Await.result(supervisorActor ? AddConnection(options, mongosystem), Duration.Inf)
+    connection.asInstanceOf[MongoConnection]
```

Comment:
The infiite timeout on the Await causes me some concern too. I'll have to revisit this.